bridge-forwarder/src/xcm_asset_transactor.rs

```diff
@@ -4,34 +4,43 @@
+use xcm::prelude::*;
+use xcm_executor::{traits::TransactAsset, Assets};
 
-pub struct XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder>(PhantomData<(CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder)>);
-impl<CurrencyTransactor: TransactAsset, FungiblesTransactor: TransactAsset, AssetTypeChecker: AssetTypeIdentifier, Forwarder: TransactorForwarder> TransactAsset for XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder> {
-    // deposit_asset implements the TransactAsset deposit_asset method and contains the logic to classify
-    // the asset recipient location:
-    // 1. recipient is on the local parachain
-    // 2. recipient is on non-substrate chain(evm, cosmos, etc.)
-    // 3. recipient is on the remote parachain
-    fn deposit_asset(what: &MultiAsset, who: &MultiLocation, context: &XcmContext) -> XcmResult {
-        match (who.parents, who.first_interior()) {
-            // 1. recipient is on the local parachain
-            (0, Some(Parachain(_))) => {
-                // check if the asset is native, and call the corresponding deposit_asset()
-                if AssetTypeChecker::is_native_asset(what) {
-                    CurrencyTransactor::deposit_asset(what, who, context)?;
-                } else {
-                    FungiblesTransactor::deposit_asset(what, who, context)?
-                }
-            }
-            // recipient is on the remote chain
-            (1, Some(Parachain(_))) => {
-                // 2. recipient is on non-substrate chain(evm, cosmos, etc.), will forward to sygma bridge pallet
-                match who.interior {
+pub struct XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder>(
+	PhantomData<(CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder)>,
+);
+impl<
+		CurrencyTransactor: TransactAsset,
+		FungiblesTransactor: TransactAsset,
+		AssetTypeChecker: AssetTypeIdentifier,
+		Forwarder: TransactorForwarder,
+	> TransactAsset
+	for XCMAssetTransactor<CurrencyTransactor, FungiblesTransactor, AssetTypeChecker, Forwarder>
+{
+	// deposit_asset implements the TransactAsset deposit_asset method and contains the logic to classify
+	// the asset recipient location:
+	// 1. recipient is on the local parachain
+	// 2. recipient is on non-substrate chain(evm, cosmos, etc.)
+	// 3. recipient is on the remote parachain
+	fn deposit_asset(what: &MultiAsset, who: &MultiLocation, context: &XcmContext) -> XcmResult {
+		match (who.parents, who.first_interior()) {
+			// 1. recipient is on the local parachain
+			(0, Some(AccountId32 { .. })) => {
+				// check if the asset is native, and call the corresponding deposit_asset()
+				if AssetTypeChecker::is_native_asset(what) {
+					CurrencyTransactor::deposit_asset(what, who, context)?;
+				} else {
+					FungiblesTransactor::deposit_asset(what, who, context)?
+				}
+			},
+			// recipient is on the remote chain
```

Comment:
that is new to me, thanks for letting me know! Based on the current condition logic, if the `parent` is 2, it will go to the `3. recipient is on remote parachain, will forward to xcm bridge pallet` branch, which I think is the correct path to go, wdyt? Maybe it is safer to add one more condition saying that if the parent is 2, we raise an error saying cross Kusama <> Polkadot bridge is not supported yet